packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,289 @@
+import * as React from 'react';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { useComposedRefs } from '@radix-ui/react-compose-refs';
+import { createContext } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as PopperPrimitive from '@radix-ui/react-popper';
+import { Portal } from '@radix-ui/react-portal';
+import { Presence } from '@radix-ui/react-presence';
+import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
+import { useId } from '@radix-ui/react-id';
+import { Slot } from '@radix-ui/react-slot';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCard
+ * -----------------------------------------------------------------------------------------------*/
+
+const HOVERCARD_NAME = 'HoverCard';
+
+type HoverCardContextValue = {
+  triggerRef: React.RefObject<HTMLAnchorElement>;
+  contentId: string;
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onMouseEnter(): void;
+  onMouseLeave(): void;
+  hasCustomAnchor: boolean;
+  onCustomAnchorAdd(): void;
+  onCustomAnchorRemove(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  enterDelayDuration?: number;
+  exitDelayDuration?: number;
```

Comment:
I know we've used `delayDuration` in `Tooltip` but I've since wondered why we need to include the word `duration` ... isn't a delay inherently a duration?

Thoughts on renaming these to `enterDelay` and `exitDelay`?

Reply:
I agree, I went with duration only for consistency.

Reply:
Yes you are right, but the reason we went with the added `duration` suffix in `Tooltip` is because of the second one `skipDelayDuration`, because otherwise `skipDelay` sounds like a boolean, or you end up with `skipDelayDelay`…

Reply:
Seems the tooltip naming means we are now having to overcomplicate our other APIs for consistency-sake. Perhaps we should revisit the tooltip naming? Happy to keep these as they are for now till we do that if so.

Reply:
Happy to revisit the tooltip naming but Colm and I spent quite a while thinking about it and that's the least worse we could come up with… But again, I would be over the moon if we can find a better idea for sure!

Reply:
I'm wondering if they should be renamed to something like `enterDelay` and `enterDelayThreshold`.

If I've not misunderstood, I believe the `skipDelayDuration` is an amount of time that the `enterDelay` will be ignored when hovering between tooltips? A threshold seems to fit that definition:

> the magnitude or intensity that must be exceeded for a certain reaction, phenomenon, result, or condition to occur or be manifested.

So, with a threshold of 400ms, it will take 400ms before the `enterDelay` will manifest between tooltips. 

If we can come up with something quick enough, maybe we can just make the change here to save increasing potential tech debt 🤷‍♀️

Reply:
btw, I'm also wondering if this should have the same thing as tooltip with the skipping delay thing.
When I see UIs like this:
![image](https://user-images.githubusercontent.com/1539897/114709885-02b99680-9d25-11eb-8edd-abf8fe901863.png)
It's quite annoying that you get the delay everytime, and that's with them having made it as short as they could probably for that reason.

Reply:
> I'm wondering if they should be renamed to something like `enterDelay` and `enterDelayThreshold`.
> 
> If I've not misunderstood, I believe the `skipDelayDuration` is an amount of time that the `enterDelay` will be ignored for when moving between tooltips? A threshold seems to fit that definition:
> 
> > the magnitude or intensity that must be exceeded for a certain reaction, phenomenon, result, or condition to occur or be manifested.
> 
> So, with a threshold of 400, it will take 400ms before the `enterDelay` will manifest between tooltips.
> 
> If we can come up with something quick enough, maybe we can just make the change here to save increasing potential tech debt 🤷‍♀️

That could work! I remember I played with "threshold" too.
I find that `enterDelayThreshold` doesn't communicate very well what it does, ie. it's hard to know if the threshold applies to the delay happening, or NOT happening.

Maybe `enterDelay` + `skipEnterDelayThreshold`?

I remember before internally it was called `SKIP_REST_THRESHOLD_DURATION`…

Reply:
> I'm also wondering if this should have the same thing as tooltip with the skipping delay thing

Yeah, that's not a bad shout tbf. @andy-hook It might be worth copying over a bunch of the state machine stuff from `Tooltip` in that case 😅 

@colmtuite What are your thoughts on this one?

Reply:
> it's hard to know if the threshold applies to the delay happening, or NOT happening.

Given the definition of "threshold" above, a threshold in this case would be the time it takes **before** the `enterDelay` would manifest. The definition of threshold already encapsulates the "skip delay for {x} ms" part in my mind. The only part that's missing for me is the fact that this only applies _between_ tooltips. 

Something like `enterDelay` and `reEnterDelayThreshold` seems to cover it. Perhaps I'll create a ticket and we can discuss this all there and rename all later if we can come up with something fitting? It's proving to be a bit harder than I'd thought 😅

Reply:
I actually really like `reEnter` that definitely brings more meaning to it :+1

Reply:
I think regardless of if that's the best we can come up with `enterDelay` + `reEnterDelayThreshold` for me is definitely better than `delayDuration` + `skipDelayDuration`!

Reply:
Ok, so @jjenzz and I continued that discussion on Slack and this is where we're at.

- `enterDelay` => `openDelay`
- `exitDelay` => `closeDelay`
> we generally use `open`/`close`, ie. css animations, `[data-state="open"]`
- `skipDelayDuration ` => `reEntryThreshold` (ie. when do we consider it a fresh re-entry, may or may not apply to `HoverCard` based on the decision we make about that feature)

So to summarise, as it stands:
- `Tooltip` would have `openDelay` + `reEntryThreshold`
- `HoverCard` would have `openDelay` + `closeDelay`

And we're considering whether `HoverCard` should have `reEntryThreshold` and its associated functionality.

Thoughts @andy-hook, @colmtuite?

Reply:
Onboard with the naming 👍 

`reEntryThreshold` functionality does sound useful for the case mentioned above, interested to hear @colmtuite thoughts.

Reply:
I love the `threshold` concept. I also love renaming to `open` and `closed`.

Not a fan of `reEntryThreshold`, we're getting into rocket science territory here 😓 I think `skipDelayThreshold` is more intuitive, though there's the issue with its meaning. We actually want an antonym for "skip", but `includeDelayThreshold` doesn't have the same ring to it. `skipDelayLimit` might work?

I think I'd vote to not include the skipDelay feature for HoverCard. In that exact GH example posted above, it would likely be beneficial. But I think in most cases, it could become quite annoying. These things can get pretty large, with many instances on the screen, like in a social feed for example. Having them open instantly as you move your mouse around could get very irritating.

Reply:
> These things can get pretty large, with many instances on the screen, like in a social feed for example. Having them open instantly as you move your mouse around could get very irritating.

Oh yeah! Good point haha. I'm on board with leaving it out for `HoverCard` too in this case.

Reply:
Great point on the size of them @colmtuite, let's leave that feature out of `HoverCard` then.
Regarding the name thing, I still prefer `reEntryThreshold` over `skipDelayDuration`.

Reply:
`openDelayThreshold`?

Reply:
we had this at some point, but we thought it doesn't convey the fact that it only applies when re-entering… I still think `reEntryThreshold` is clearer for that reason. ie. the time is about how do I have to re-enter.
The biggest issue with something like `openDelayThreshold` is that `delay` and `threshold` are kinda similar.

Reply:
I suppose this discussion is now resolved from the POV of `HoverCard`. We can reopen when we look at aligning `Tooltip`'s API.